### PR TITLE
Removed unneeded checks for passing null to free()

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

File: libusb/io.c [hunk 1]
```diff
@@ -2755,15 +2752,12 @@ const struct libusb_pollfd ** LIBUSB_CALL libusb_get_pollfds(
  * Since version 1.0.20, \ref LIBUSB_API_VERSION >= 0x01000104
  *
  * It is legal to call this function with a NULL pollfd list. In this case,
- * the function will simply return safely.
+ * the function will simply no nothing.
```
Comment (LudovicRousseau) [inline libusb/io.c]: I guess a "return" word is missing here.

File: libusb/os/darwin_usb.c [hunk 2]
```diff
@@ -1675,8 +1675,8 @@ static int submit_iso_transfer(struct usbi_transfer *itransfer) {
 
   struct darwin_interface *cInterface;
 
-  /* construct an array of IOUSBIsocFrames, reuse the old one if possible */
-  if (tpriv->isoc_framelist && tpriv->num_iso_packets != transfer->num_iso_packets) {
+  /* construct an array of IOUSBIsocFrames, reuse the old one if the sizes are the same */
+  if (tpriv->num_iso_packets != transfer->num_iso_packets) {
```
Comment (LudovicRousseau) [inline libusb/os/darwin_usb.c]: This change should NOT be part of the same PR.

File: libusb/io.c (hunk 1 above)
Comment (seanm) [reply libusb/io.c]: Whoops, I meant "do". :)

File: libusb/os/darwin_usb.c (hunk 2 above)
Comment (seanm) [reply libusb/os/darwin_usb.c]: Why? It's the same thing: removing a check for null before calling free().

Comment (LudovicRousseau) [reply libusb/os/darwin_usb.c]: Exact. Sorry.

File: libusb/io.c (hunk 1 above)
Comment (seanm) [reply libusb/io.c]: Fixed.

**Comment by seanm** [comment]
So with the comment fixed, will you be merging this?

**Comment by hjelmn** [comment]
Looks good to me.

**Comment by LudovicRousseau** [comment]
ok for me

**Comment by seanm** [comment]
So is there something I need to do here?  I don't see any button that would allow me to merge it...

**Comment by hjelmn** [comment]
@seanm We do this all by hand to make sure the nano gets updated. I plan to go through all the PRs today or tomorrow.

**Pull request closed without merging**